script/output/scroll_sepolia/Groth16VerifierExtension.sol.ignore

```diff
@@ -0,0 +1,280 @@
+            );
+        }
+
+        // TODO: Uncomment once limit and offset supported
+        // Check the query limit and offset.
+        // uint32 limit = uint32(bytes4(rem << (REM_QUERY_LIMIT_POS * 32)));
+        // require(limit == query.limit, "Query limit must equal as expected.");
+        // uint32 offset = uint32(bytes4(rem << (REM_QUERY_OFFSET_POS * 32)));
+        // require(offset == query.offset, "Query offset must equal as expected.");
+
+        // Throw an error if overflow.
+        uint32 overflow = uint32(bytes4(rem << (REM_OVERFLOW_POS * 32)));
+        if (overflow == 0) {
+            return QueryErrorCode.NoError;
+        }
+        return QueryErrorCode.ComputationOverflow;
+    }
+
+    /// @notice verifies two blockhashed are equal
+    /// @param blockHash the blockhash computed from the proof
+    /// @param expectedBlockHash the expected blockhash, retrieved from the query
+    /// @dev this function is virtual to allow for different implementations in different environments
+    function verifyBlockHash(bytes32 blockHash, bytes32 expectedBlockHash) internal view virtual {
+        require(blockHash == expectedBlockHash, "Block hash must equal as expected.");
+    }
+
+    // Parse the query output from the public inputs.
+    function parseOutput(bytes32[] calldata data, QueryErrorCode error)
+        internal
+        pure
+        virtual
+        returns (QueryOutput memory)
+    {
+        bytes32 rem = data[PI_REM_OFFSET];
+
+        // Retrieve total number of the matched rows.
+        uint32 totalMatchedRows = uint32(bytes4(rem << (REM_ENTRY_COUNT_POS * 32)));
+
+        // Retrieve the current result number.
+        uint32 numResults = uint32(bytes4(rem << (REM_NUM_RESULTS_POS * 32)));
+        require(numResults <= MAX_NUM_OUTPUTS, "Result number cannot overflow.");
+
+        uint32 offset = PI_OFFSET + RESULT_VALUES_POS;
+        bytes[] memory rows = new bytes[](numResults);
+
+        for (uint32 i = 0; i < numResults; ++i) {
+            uint256[] memory columns = new uint256[](MAX_NUM_ITEMS_PER_OUTPUT);
+            for (uint32 j = 0; j < MAX_NUM_ITEMS_PER_OUTPUT; ++j) {
+                columns[j] = uint256(data[offset + i * MAX_NUM_ITEMS_PER_OUTPUT + j]);
+            }
+            rows[i] = abi.encodePacked(columns);
+        }
+
+        QueryOutput memory output = QueryOutput({totalMatchedRows: totalMatchedRows, rows: rows, error: error});
+
+        return output;
+    }
+
+    // Reverse the bytes of each Uint32 in block hash.
+    // Since we pack to little-endian for each Uint32 in block hash.
+    function convertToBlockHash(bytes32 original) internal pure returns (bytes32) {
+        bytes32 result;
+        for (uint256 i = 0; i < 8; ++i) {
+            for (uint256 j = 0; j < 4; ++j) {
+                result |= bytes32(original[i * 4 + j]) >> (8 * (i * 4 + 3 - j));
+            }
+        }
+
+        return result;
+    }
+}
```

Comment:
This is the file that has been generated for Scroll ? 
I thought we said one "environement" could be deployed on multiple chains,in this case it should be the same file as we have on L1 Holesky

Reply:
Yes, this is the file that was generated for Scroll. It will be identical to all other "test" env chains generated with the same version of the `copy-verifier` script. This file does not really need to be checked into version control and we could just have one for each environment rather than one for each chain, but for the sake of simplicity it was easier to just save one copy per chain for now. Additionally, this contract should really be the same for _all_ deployments across chain, across environment. It does not vary from chain to chain or env to env, which is why I requested we pull it into the main contracts repo [here](https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/412).

This contract is _not_ identical to any of the existing Holesky deployments because I made changes to `Groth16VerifierExtension.sol` [here](https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/414). These changes were made in a backwards-compatibly manner, so there is no need to redeploy contracts to make them match perfectly b/t chains.